pkg/debug/start_debug.go

```diff
@@ -0,0 +1,162 @@
+	if err != nil {
+		fmt.Printf("Missing mon or osd deployment name %s. %v\n", deploymentName, err)
+		os.Exit(1)
+	}
+
+	if alternateImageValue != "" {
+		deployment.Spec.Template.Spec.Containers[0].Image = alternateImageValue
+	}
+
+	labels := deployment.Labels
+	labels["ceph.rook.io/do-not-reconcile"] = "true"
+
+	deployment.Spec.Template.Spec.Containers[0].LivenessProbe = nil
+	deployment.Spec.Template.Spec.Containers[0].StartupProbe = nil
+
+	if alternateImageValue != "" {
+		log.Printf("setting debug image to %s\n", alternateImageValue)
+		deployment.Spec.Template.Spec.Containers[0].Image = alternateImageValue
+	}
+
+	fmt.Println("setting debug command to main container")
+
+	deployment.Spec.Template.Spec.Containers[0].Command = []string{"sleep", "infinity"}
+	deployment.Spec.Template.Spec.Containers[0].Args = []string{}
+
+	if err := updateDeployment(context, clusterNamespace, deployment); err != nil {
+		fmt.Printf("Failed to update deployment %s. %v\n", deployment.Name, err)
+		os.Exit(1)
+	}
+
+	deploymentPodName, err := podRunningWithLabel(context, clusterNamespace, deployment.Spec)
+	if deploymentPodName == "" || err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	if err := setDeploymentScale(context, clusterNamespace, deployment.Name, 0); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("waiting for the deployment pod %s to be deleted\n", deploymentPodName)
+
+	err = waitForPodDeletion(context, clusterNamespace, deploymentPodName)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	debugDeploymentSpec := &appsv1.Deployment{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-debug", deploymentName),
+			Namespace: clusterNamespace,
+			Labels:    labels,
+		},
+		Spec: deployment.Spec,
+	}
+
+	debugDeployment, err := context.Clientset.AppsV1().Deployments(clusterNamespace).Create(ctx.TODO(), debugDeploymentSpec, v1.CreateOptions{})
+	if err != nil {
+		fmt.Printf("Error creating deployment %s. %v\n", debugDeploymentSpec, err)
+		os.Exit(1)
+	}
+	fmt.Printf("ensure the debug deployment %s is scaled up\n", deploymentName)
+
+	if err := setDeploymentScale(context, clusterNamespace, debugDeployment.Name, 1); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+}
+
+func setDeploymentScale(context *k8sutil.Context, clusterNamespace, deploymentName string, scaleCount int) error {
+	scale := &autoscalingv1.Scale{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      deploymentName,
+			Namespace: clusterNamespace,
+		},
+		Spec: autoscalingv1.ScaleSpec{
+			Replicas: int32(scaleCount),
+		},
+	}
+	_, err := context.Clientset.AppsV1().Deployments(clusterNamespace).UpdateScale(ctx.TODO(), deploymentName, scale, v1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to update scale of deployment %s. %v\n", deploymentName, err)
+	}
+	return nil
+}
+
+func verifyDeploymentExists(context *k8sutil.Context, clusterNamespace, deploymentName string) (*appsv1.Deployment, error) {
```

Comment:
I'd suggest no need for helper methods like this were it's such a simple wrapper around a simple method. How about just calling `Deployments.Get()` where it's needed instead of using the helper? Same suggestion for `updateDeployment()` below

Reply:
I tried to make in a helper function as, these methods will be used commonly in other places, and in the future, if we implement mon recovery it will be required too.

I can still remove it if you suggest.